### PR TITLE
Create Log4J IP Detections and create exploit list

### PR DESCRIPTION
**Pull request by bseb**
### Background

- Creates IOC list of Log4J exploit strings and known IPs that have been seen exploiting this CVE in the wild
- Creates rule to detect IPs in various log sources

PR 1 of 2(?). In the interest of this PR not becoming massive I'm going to create a helper and rules for detecting the exploit strings on the next PR

### Changes

* See Above

### Testing

* CI/CD, unit tests

### CODE REVIEW

**Comment by k-bailey** [comment]
Looks good to me, I don't have codeowner status but approved

**Review by darwayne — APPROVED** [review]

File: panther_ioc_rules/log4j_ip_iocs.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - p_any_ip_addresses
+Tests:
+  -
+    Name: Non-matching traffic
+    ExpectedResult: false
+    Log:
+      {
+        "dstport": 53,
+        "dstaddr": "1.1.1.1",
+        "srcaddr": "10.0.0.1"
```
Comment (lindsey-w) [inline panther_ioc_rules/log4j_ip_iocs.yml:46]: perhaps add `p_any_ip_addresses` field just for confirmation that this doesn't match

**Review by lindsey-w — APPROVED** [review]

File: global_helpers/panther_iocs.py [hunk 2]
```diff
@@ -34,7 +34,663 @@
...
+    "185.220.100.252",
+    "72.223.168.73",
+    "51.255.106.85",
+    "46.182.21.248",
+    "23.129.64.148",
+    "23.129.64.146",
+    "23.129.64.131",
+     }
+
+LOG4J_EXPLOIT_IOCS = {
```
Comment (darwayne) [inline global_helpers/panther_iocs.py:689]: assuming this will be used in follow up PR.

Comment (bseb) [reply global_helpers/panther_iocs.py:689]: Correct

File: global_helpers/panther_iocs.py [hunk 3]
```diff
@@ -34,7 +34,663 @@
     "d0d626deb3f9484e649294a8dfa814c5568f846d5aa02d4cdad5d041a29d5600",
 }
 
+#LOG4J IOCs:
+# IPs Pulled from Greynoise and compiled here:
+# https://gist.github.com/gnremy/c546c7911d5f876f263309d7161a7217
+# Created 12-13-21
 
+LOG4J_IP_IOCS = {
+    "167.99.88.151",
```
Comment (jpl5280) [inline global_helpers/panther_iocs.py:43]: sort?

File: global_helpers/panther_iocs.py [hunk 4]
```diff
@@ -34,7 +34,663 @@
...
+    "72.223.168.73",
+    "51.255.106.85",
+    "46.182.21.248",
+    "23.129.64.148",
+    "23.129.64.146",
+    "23.129.64.131",
+     }
+
+LOG4J_EXPLOIT_IOCS = {
+     'jndi:ldap:/', 'jndi:rmi:/', 'jndi:ldaps:/', 'jndi:dns:/', 'jndi:nis:/', 'jndi:nds:/', 'jndi:corba:/', 'jndi:iiop:/'
```
Comment (jpl5280) [inline global_helpers/panther_iocs.py:690]: RE https://twitter.com/James_inthe_box/status/1470408273727668225?s=20, should we do `jndi:${l`

Comment (jpl5280) [reply global_helpers/panther_iocs.py:690]: Keep in mind, this won't catch https://twitter.com/Rezn0k/status/1469523006015750146?s=20

File: panther_ioc_rules/log4j_ip_iocs.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      }
+  -
+    Name: Sunburst Indicator of Compromise (IP) Detected
```
Comment (jpl5280) [inline panther_ioc_rules/log4j_ip_iocs.yml:49]: needs to be updated?

Comment (bseb) [reply panther_ioc_rules/log4j_ip_iocs.yml]: Updated. Used the Sunburst yml as a template and forgot to change it :shame:

File: panther_ioc_rules/log4j_ip_iocs.yml (hunk 1 above)
Comment (bseb) [reply panther_ioc_rules/log4j_ip_iocs.yml]: Done :)

File: global_helpers/panther_iocs.py [hunk 6]
```diff
@@ -34,7 +34,663 @@
     "d0d626deb3f9484e649294a8dfa814c5568f846d5aa02d4cdad5d041a29d5600",
 }
 
+#LOG4J IOCs:
+# IPs Pulled from Greynoise and compiled here:
+# https://gist.github.com/gnremy/c546c7911d5f876f263309d7161a7217
+# Created 12-13-21
 
+LOG4J_IP_IOCS = {
```
Comment (jpl5280) [inline global_helpers/panther_iocs.py:42]: would suggest adding a comment of where you got it from? Here's some others

https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Sample%20Data/Feeds/Log4j_IOC_List.csv
https://github.com/Malwar3Ninja/Exploitation-of-Log4j2-CVE-2021-44228/blob/main/Threatview.io-log4j2-IOC-list

File: global_helpers/panther_iocs.py (hunk 4 above)
Comment (bseb) [reply global_helpers/panther_iocs.py:690]: > should we do jndi:${ 
Yeah thats a good call for sure, adding

> Tangentially, keep in mind, this won't catch all possible permutations
The amount of WAF / detection bypasses I'm seeing for this is insane

Comment (bseb) [reply global_helpers/panther_iocs.py:690]: Also adding `:${lower:`

File: global_helpers/panther_iocs.py (hunk 6 above)
Comment (bseb) [reply global_helpers/panther_iocs.py:42]: https://github.com/panther-labs/panther-analysis/pull/356/files#diff-56771479b9258fa730e354c5959c7c1228ef8a7d5863a78db8f48e0f48ef93b5R38

File: global_helpers/panther_iocs.py [hunk 7]
```diff
@@ -34,7 +34,663 @@
...
+        "1.116.59.211",
+"1.14.17.89",
```
Comment (darwayne) [inline global_helpers/panther_iocs.py:44]: `[minor]` looks like your formatting is off with the new update

File: global_helpers/panther_iocs.py (hunk 6 above)
Comment (bseb) [reply global_helpers/panther_iocs.py:44]: Added and deduped

**Review by darwayne — APPROVED** [review]

**Review by lindsey-w — APPROVED** [review]

**Pull request merged**